nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-cluster/src/main/java/org/apache/nifi/cluster/coordination/http/replication/ThreadPoolRequestReplicator.java

```diff
@@ -638,21 +668,37 @@ private boolean isMutableRequest(final String method) {
         }
     }
 
-    private boolean isDeleteConnection(final String method, final String uriPath) {
+    private boolean isDeleteComponent(final String method, final String uriPath) {
         if (!HttpMethod.DELETE.equalsIgnoreCase(method)) {
             return false;
         }
 
-        final boolean isConnectionUri = ConnectionEndpointMerger.CONNECTION_URI_PATTERN.matcher(uriPath).matches();
-        return isConnectionUri;
+        // Check if the URI indicates that a component should be deleted.
+        // We cannot simply make our decision based on the fact that the request is a DELETE request.
+        // This is because we do need to allow deletion of asynchronous requests, such as updating parameters, querying provenance, etc.
+        // which create a request, poll until the request completes, and then deletes it. Additionally, we want to allow terminating
+        // Processors, which is done by issuing a request to DELETE /processors/<id>/threads
```

Comment:
This approach is understandable, but it functions as a list of blocked request patterns. Would it make more sense to reverse the logic to check against a list of "non-component" patterns so that it functions as a list of allowed request patterns? With that approach, unless the operation is explicitly listed, it will be cause the `IllegalClusterStateException`, which seems a bit safer in the event of new component types added to the framework. Not sure of the scope without further evaluation, but wanted to raise the question.